src/DiscreteFilter.hpp

```diff
@@ -0,0 +1,134 @@
+#ifndef DISCRETEFILTER_HPP_
+#define DISCRETEFILTER_HPP_
+
+#include <base/samples/Joints.hpp>
+#include <base/samples/RigidBodyState.hpp>
+#include <iostream>
+
+namespace base {
+
+/**
+ * 	This library implements a second order low pass discrete filter from a continuous filter using the matched pole-zero methods.
+ *
+ *		   (CONTINUOUS FILTER)						    (DISCRETE FILTER)
+ *			        kc								           kd
+ *	G(s) =  -----------------   ======>  G(z) = --------------------------------
+ *			 (s + p1)(s + p2)					 (1 - ep1T*z^⁻1)(1 - ep2T*z^-2)
+ *
+ *  where:
+ *
+ *	- epT: Continuous pole matched in discrete time ===> epT = exp(p*T)
+ *	- p:   Continuous pole
+ *	- T:   Sample time
+ *	- kd:  Unitary static gain						===> kd = 1 - (ep1T + ep2T) + ep1T*ep2T
+ *
+ *  The resulting difference equation is:
+ *
+ *	y[k] = kd*u[k] + (ep1T + ep2T)*y[k-1] - ep1T*ep2T*y[k-2]
+ *
+ *  This difference equation is implemented in this library to filter a specified input signal.
+ *
+ *  The user may worry about setting only the pole1, because the pole2 will copy the
+ *  pole1's value if pole2's value is not set. The value of pole1 must be greater than zero,
+ *  and basically, the bigger pole1's value the more intense is the filtering (the cutting
+ *  frequency is lower).
+ *
+ */
+
+/**
+ * Structure for the signalType parameter
+ */
+enum State
```

Comment:
Defining a base::State does not feel as a very good idea, provide a clearer naming sceme please